[GTK] MiniBrowser should hide the toolbar when using --full-screen #17909

eworm-de opened the pull request:
#### e07345343415dd2496edc721daa61a3b42703131
<pre>
[GTK] MiniBrowser should hide the toolbar when using --full-screen #17909
<a href="https://bugs.webkit.org/show_bug.cgi?id=261732">https://bugs.webkit.org/show_bug.cgi?id=261732</a>

Reviewed by Michael Catanzaro.

Entering fullscreen mode with F11 works as expected.
Starting with command line switch `--full-screen` does make the window
fullscreen, but does not hide the toolbar. Let&apos;s change that.

Instead of just making the window fullscreen with `gtk_window_fullscreen()`
this introduces a new function `browser_window_fullscreen()` that hides
the toolbar as well.

Also introduce new function `browserWindowUnfullscreen()`, use both
in `toggleFullScreen()`, and drop extra inversion.

* Tools/MiniBrowser/gtk/BrowserWindow.c
* Tools/MiniBrowser/gtk/BrowserWindow.h
* Tools/MiniBrowser/gtk/main.c

Canonical link: <a href="https://commits.webkit.org/268141@main">https://commits.webkit.org/268141@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4ca126adef66f6b544ccb3b167e0d12f9f2083e4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18834 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19177 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19781 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20703 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17631 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19028 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22482 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19315 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19425 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19060 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19195 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16405 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21585 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16415 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17167 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/23601 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17444 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17339 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21513 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/17935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17000 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4467 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21367 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/17771 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (changes requested):
So, `_BrowserWindow` is the private data struct for `BrowserWindow`. Making it public like this allows other code to access the internals of the `BrowserWindow`. So that's not good! Instead, leave the struct where it is and add a new function `browser_window_fullscreen()` to do this work from BrowserWindow.c.

Does it work properly when you return from fullscreen?

eworm-de reviewed (commented):
Changes applied.

mcatanzaro reviewed (changes requested):
OK, I actually found two more things we can improve.

Normally I would approve of putting the new `browser_window_fullscreen()` function right above `browser_window_load_uri()` in BrowserWindow.c to match where you've got it located in BrowserWindow.h. But in this particular case, I think it's actually safer to put it right above `webViewEnterFullScreen` instead. This way, we keep the fullscreen-related code in one place and it's less likely to break in the future because anybody editing one function or the other should see that the other one exists.

Next, I'd like `toggleFullScreen()` to call the new `browser_window_fullscreen()` to reduce code duplication. For parallelism, you can add a new `browserWindowUnfullscreen()` function as well. For example:

```
static void toggleFullScreen(GSimpleAction *action, GVariant *parameter, gpointer userData)
{
    BrowserWindow *window = BROWSER_WINDOW(userData);
    if (!window->fullScreenIsEnabled)
        browser_window_fullscreen(window);
    else
        browserWindowUnfullscreen(window);
}
```

(The public functions use underscores, while the private functions use camelCase.)

mcatanzaro commented on Tools/MiniBrowser/gtk/BrowserWindow.c:
```diff
@@ -482,6 +482,20 @@ static GtkWidget *webViewCreate(WebKitWebView *webView, WebKitNavigationAction *
+    window->fullScreenIsEnabled = TRUE;
+}
+
+void browserWindowUnfullscreen(BrowserWindow *window)
```
OK, one last change: this file-scope function should not be exported:

```suggestion
static void browserWindowUnfullscreen(BrowserWindow *window)
```

mcatanzaro reviewed (commented):
Thanks for updating the pull request. This looks good except for the one-line fix I've suggested. Please leave another comment after you've updated the pull request one last time so I know to merge it. Thanks!

eworm-de committed:
[GTK] MiniBrowser should hide the toolbar when using --full-screen #17909
https://bugs.webkit.org/show_bug.cgi?id=261732

Reviewed by Michael Catanzaro.

Entering fullscreen mode with F11 works as expected.
Starting with command line switch `--full-screen` does make the window
fullscreen, but does not hide the toolbar. Let's change that.

Instead of just making the window fullscreen with `gtk_window_fullscreen()`
this introduces a new function `browser_window_fullscreen()` that hides
the toolbar as well.

Also introduce new function `browserWindowUnfullscreen()`, use both
in `toggleFullScreen()`, and drop extra inversion.

* Tools/MiniBrowser/gtk/BrowserWindow.c
* Tools/MiniBrowser/gtk/BrowserWindow.h
* Tools/MiniBrowser/gtk/main.c

Canonical link: https://commits.webkit.org/268141@main
